packages/cli/src/modules/insights/database/repositories/insights-by-period.repository.ts

```diff
@@ -228,4 +254,89 @@ export class InsightsByPeriodRepository extends Repository<InsightsByPeriod> {
 
 		return summaryParser.parse(rawRows);
 	}
+
+	private parseSortingParams(sortBy: string): [string, 'ASC' | 'DESC'] {
+		const [column, order] = sortBy.split(':');
+		return [column, order.toUpperCase() as 'ASC' | 'DESC'];
+	}
+
+	async getInsightsByWorkflow({
+		nbDays,
+		skip = 0,
+		take = 20,
+		sortBy = 'total:desc',
+	}: {
+		nbDays: number;
+		skip?: number;
+		take?: number;
+		sortBy?: string;
+	}) {
+		const dateSubQuery =
+			dbType === 'sqlite'
+				? `datetime('now', '-${nbDays} days')`
+				: dbType === 'postgresdb'
+					? `NOW() - INTERVAL '${nbDays} days'`
+					: `DATE_SUB(NOW(), INTERVAL ${nbDays} DAY)`;
+
+		const [sortField, sortOrder] = this.parseSortingParams(sortBy);
+		const sumOfExecutions = sql`SUM(CASE WHEN insights.type IN (${TypeToNumber.success.toString()}, ${TypeToNumber.failure.toString()}) THEN value ELSE 0 END)`;
+
+		const rawRowsQuery = this.createQueryBuilder('insights')
+			.select([
+				'metadata.workflowId AS "workflowId"',
+				'metadata.workflowName AS "workflowName"',
+				'metadata.projectId AS "projectId"',
+				'metadata.projectName AS "projectName"',
+				`SUM(CASE WHEN insights.type = ${TypeToNumber.success} THEN value ELSE 0 END) AS "succeeded"`,
+				`SUM(CASE WHEN insights.type = ${TypeToNumber.failure} THEN value ELSE 0 END) AS "failed"`,
+				`SUM(CASE WHEN insights.type IN (${TypeToNumber.success}, ${TypeToNumber.failure}) THEN value ELSE 0 END) AS "total"`,
+				sql`CASE
+								WHEN ${sumOfExecutions} = 0 THEN 0
+								ELSE 1.0 * SUM(CASE WHEN insights.type = ${TypeToNumber.failure.toString()} THEN value ELSE 0 END) / ${sumOfExecutions}
+							END AS "failureRate"`,
+				`SUM(CASE WHEN insights.type = ${TypeToNumber.runtime_ms} THEN value ELSE 0 END) AS "runTime"`,
+				`SUM(CASE WHEN insights.type = ${TypeToNumber.time_saved_min} THEN value ELSE 0 END) AS "timeSaved"`,
+				sql`CASE
+								WHEN ${sumOfExecutions} = 0	THEN 0
+								ELSE 1.0 * SUM(CASE WHEN insights.type = ${TypeToNumber.runtime_ms.toString()} THEN value ELSE 0 END) / ${sumOfExecutions}
+							END AS "averageRunTime"`,
+			])
+			.innerJoin('insights.metadata', 'metadata')
+			.where(`insights.periodStart >= ${dateSubQuery}`)
+			.groupBy('metadata.workflowId')
+			.addGroupBy('metadata.workflowName')
+			.addGroupBy('metadata.projectId')
+			.addGroupBy('metadata.projectName')
```

Comment:
If we group by `workflowId` and every workflow can only exist in in project and have one name, do we need to group by these too?

Reply:
@despairblue since those fields are in the select statement, some db engines (like mysql with ONLY_FULL_GROUP_BY true) require all non aggregated selected to have a matching group by statement.
I could also have used `any_value`, but I think it does not exist in sqlite, which would add `if`s

Reply:
TIL, thanks, seems postgres by default is really specific about this.